L1Trigger/DTTriggerPhase2/src/MuonPathAssociator.cc

```diff
@@ -400,25 +402,46 @@ void MuonPathAssociator::correlateMPaths(edm::Handle<DTDigiCollection> dtdigis,
                   lat = 0;
                 }
                 if (std::abs(x_inSL3 - x_wire) < minx) {
-                  min2x = minx;
-                  minx = std::abs(x_inSL3 - x_wire);
-                  next_wire = best_wire;
-                  next_tdc = best_tdc;
-                  next_layer = best_layer;
-                  next_lat = best_lat;
-
-                  best_wire = (*digiIt).wire();
-                  best_tdc = (*digiIt).time();
-                  best_layer = dtLId.layer();
-                  best_lat = lat;
-                  matched_digis++;
+                  // different layer than the stored in best, hit added, matched_digis++;. This approach in somewhat
+                  // buggy, as we could have stored as best LayerX -> LayerY -> LayerX, and this should
+                  // count only as 2 hits. However, as we confirm with at least 2 hits, having 2 or more
+                  // makes no difference
+                  if (dtLId.layer() != best_layer) {
+                    minx = std::abs(x_inSL3 - x_wire);
+                    next_wire = best_wire;
+                    next_tdc = best_tdc;
+                    next_layer = best_layer;
+                    next_lat = best_lat;
+
+                    best_wire = (*digiIt).wire();
+                    best_tdc = (*digiIt).time();
+                    best_layer = dtLId.layer();
+                    best_lat = lat;
+                    matched_digis++;
+                  } else if (dtLId.layer() ==
+                             best_layer) {  // same layer than stored, just substituting the hit, no matched_digis++;
+                    best_wire = (*digiIt).wire();
+                    best_tdc = (*digiIt).time();
+                    best_layer = dtLId.layer();
+                    best_lat = lat;
+                  }
                 } else if ((std::abs(x_inSL3 - x_wire) >= minx) && (std::abs(x_inSL3 - x_wire) < min2x)) {
+                  // same layer than the stored in best, no hit added
+                  if (dtLId.layer() == best_layer)
+                    continue;
+                  // different layer than the stored in next, hit added. This approach in somewhat
+                  // buggy, as we could have stored as next LayerX -> LayerY -> LayerX, and this should
+                  // count only as 2 hits. However, as we confirm with at least 2 hits, having 2 or more
+                  // makes no difference
+                  else if (dtLId.layer() != next_layer)
```

Comment:
Why is the` else if` needed after a `continue` in case of the complementary possibility?